0021-zarf-values/README.md

```diff
@@ -0,0 +1,291 @@
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to specify values files for a Zarf package **so that** I can simplify package configurations and rely on my existing familiarity with Helm.
+
+**Given** I have a Zarf Package with a Helm value override in a chart
+```yaml
+components:
+  - name: component
+    charts:
+      - name: mychart
+        version: 0.1.0
+        namespace: zarf
+        localPath: chart
+        valuesFiles:
+          - values.yaml
+        values:
+          - key: component.resources
+            path: resources
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+**Given** I have a Zarf Package with Go templating inside an action
+```yaml
+components:
+  - name: component
+    actions:
+      onDeploy:
+        before:
+          - cmd: "echo \"${{ .Values.component.resources.limits.memory }}\""
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will template the action and output `128Mi`
+
+**Given** I have a Zarf Package with a setValues action
+```yaml
+components:
+  - name: component
+    actions:
+      onDeploy:
+        before:
+          - cmd: "echo '{ \"memory\": \"256Mi\", \"cpu\": \"200m\" }'"
+            setValues:
+              - path: component.resources.limits
+                type: json
+                
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will initially use the values provided for any components before the one with `setValues`
+**And Then** apply the values provided in the `setValues` action for any following components based on the actions lifecycle.
+
+> [!NOTE]
+> *This would apply to all `zarf-config` formats not just YAML
```

Comment:
This is saying all zarf-config formats could set the `package.deploy.values` key, but the values files always have to be yaml correct?

Reply:
correct - I added a line in the design details stating that explicitly